app/assets/javascripts/discourse/app/routes/tag-show.js

```diff
@@ -23,100 +23,112 @@ export default DiscourseRoute.extend(FilterModeMixin, {
 
   queryParams,
 
-  renderTemplate() {
+  controllerName: "tags.show",
+  templateName: "tags.show",
+
+  beforeModel() {
     const controller = this.controllerFor("tag.show");
-    this.render("tags.show", { controller });
+    controller.setProperties({
+      loading: true,
+      showInfo: false,
+    });
   },
 
-  model(params) {
+  async model(params, transition) {
     const tag = this.store.createRecord("tag", {
       id: escapeExpression(params.tag_id),
     });
+
+    let additionalTags;
+
     if (params.additional_tags) {
-      this.set(
-        "additionalTags",
-        params.additional_tags.split("/").map((t) => {
-          return this.store.createRecord("tag", {
-            id: escapeExpression(t),
-          }).id;
-        })
-      );
-    } else {
-      this.set("additionalTags", null);
+      additionalTags = params.additional_tags.split("/").map((t) => {
```

Comment:
is it possible for `params.additional_tags` to be an empty string?

Reply:
I don't think so. It's populated by [the ember router](https://github.com/discourse/discourse/blob/d678ba11030e77808d59c0d8d425286cf399447e/app/assets/javascripts/discourse/app/routes/app-route-map.js#L241-L241). If the param was empty, the route wouldn't match, and so `additional_tags` would be undefined.